system-helper/flatpak-system-helper.c

```diff
@@ -1151,6 +1289,215 @@ handle_run_triggers (FlatpakSystemHelper   *object,
+  g_subprocess_launcher_take_fd (launcher, sockets[0], 3);
+  target_socket = g_strdup ("--socket=3");
+  pull->revokefs_backend = g_subprocess_launcher_spawn (launcher,
+                                                        error,
+                                                        revokefs_fuse_path,
+                                                        "--backend",
+                                                        target_socket, src, NULL);
+  if (pull->revokefs_backend == NULL)
+    {
+      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_SPAWN_FAILED,
+                   "Could not spawn revokefs backend");
+      return NULL;
+    }
+
+  pull->client_socket = sockets[1];
+  return pull;
+}
+
+static gboolean
+handle_revokefs_fd (FlatpakSystemHelper   *object,
+                    GDBusMethodInvocation *invocation,
+                    GUnixFDList           *arg_fdlist,
+                    guint                  arg_flags,
+                    const gchar           *arg_ref,
+                    const gchar           *arg_commit,
+                    const gchar           *arg_installation)
+{
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GFile) mnt_file = NULL;
+  g_autoptr(GUnixFDList) fd_list = NULL;
+  g_autoptr(GError) error = NULL;
+  g_auto(GStrv) parts = NULL;
+  g_autofree gchar *mnt = NULL;
+  g_autofree gchar *src = NULL;
+  g_autofree gchar *flatpak_dir = NULL;
+  g_autofree gchar *repo_tmp = NULL;
+  g_autofree gchar *cache_dir_basename = NULL;
+  g_autofree gchar *revokefs_fuse_path = NULL;
+  struct passwd *passwd;
+  OngoingPull *new_pull;
+  uid_t uid;
+  int fd_index = -1;
+  int exit_status= -1;
+
+  g_debug ("RevokefsFd %u %s %s %s", arg_flags, arg_ref, arg_commit, arg_installation);
+
+  system = dir_get_system (arg_installation, get_sender_pid (invocation), &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  revokefs_fuse_path = g_find_program_in_path ("revokefs-fuse");
+  if (revokefs_fuse_path == NULL)
+    {
+      g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_NOT_SUPPORTED,
+                                             "Can't find bindfs in the system.");
+      return TRUE;
+    }
+
+  if (!get_connection_uid (invocation, &uid, &error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  flatpak_dir = g_file_get_path (flatpak_dir_get_path (system));
+  repo_tmp = g_build_filename (flatpak_dir, "repo", "tmp", NULL);
+
+  /* TODO: Plug in pull recovery code path */
+
+  /* Create backing directory and let it owned by "flatpak" user */
+  parts = flatpak_decompose_ref (arg_ref, NULL);
+  cache_dir_basename = g_strdup_printf ("flatpak-cache-%s-XXXXXX", parts[1]);
+  src = g_mkdtemp_full (g_build_filename (repo_tmp, cache_dir_basename, NULL), 0755);
+  passwd = getpwnam ("flatpak");
```

Comment:
error handling if the user doesn't exist

Reply:
Perhaps do this lookup once when the daemon starts?